src/helpers/pivot/pivot_helpers.ts

```diff
@@ -128,11 +128,11 @@ const DATE_FIELDS = ["date", "datetime"];
  * e.g "create_date:month" => { name: "create_date", granularity: "month" }
  */
 export function parseDimension(dimension: string): PivotCoreDimension {
```

Comment:
Shouldn't `parseDimension` completely removed ?

Some leftover uses that I think are wrong:

- It is still used in the pivot layout configurator. It should be based on the id instead of the `nameWithranularity` (see `draggableIds`)
- in `parseArgsToPivotDomain` (and `getTypeOfDimension`): `fieldWithGranularity` is coming from the formula and is actually an id.
- `SpreadsheetPivotTable` manipulates `fieldWithGranularity`, it should be id (it needs to get the actual measure from the id but it doesn't have access to the definition atm)

Reply:
> It is still used in the pivot layout configurator. It should be based on the id instead of the nameWithranularity (see draggableIds)

No, col and row does not have `id` (they should, but in a future task ?)

> in parseArgsToPivotDomain (and getTypeOfDimension): fieldWithGranularity is coming from the formula and is actually an id.

It's for col and row, so it's correct

> SpreadsheetPivotTable manipulates fieldWithGranularity, it should be id (it needs to get the actual measure from the id but it doesn't have access to the definition atm)

For measure, it's useless to get the measure, the value is always a string (the field name of the measure). So no need to update `SpreadsheetPivotTable` :)